pkg/resource/bucket/hook.go

```
@@ -74,25 +84,32 @@ func (rm *resourceManager) createPutFields(
 	ctx context.Context,
 	r *resource,
 ) error {
+	// Check if this is a directory bucket to skip unsupported operations
```

We don't want to silently ignore values in the spec field. If a user provides an invalid combination of values in their spec we want them to get feedback that the controller isn't able apply the desired state. Often this just involves calling the API and surfacing the returned validation error.

---

That makes sense. I modified it to use a validation function that returns error if the type of bucket and properties provided are incompatible